sdcm/sct_runner.py

```diff
@@ -31,24 +36,17 @@ class SctRunner:
     IMAGE_NAME = f"sct-runner-{VERSION}"
     NODE_TYPE = "sct-runner"
     RUNNER_NAME = "SCT-Runner"
-    BASE_IMAGE_ID = "ami-0ffac660dd0cb2973"  # ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20200609
-    SOURCE_IMAGE_REGION = "eu-west-2"  # where the source Runner image will be created and copied to other regions
     LOGIN_USER = "ubuntu"
-    IMAGE_DESCRIPTION = "SCT runner image"
+    IMAGE_DESCRIPTION = f"SCT runner image {VERSION}"
 
-    def __init__(self, region_name: str):
+    def __init__(self, cloud_provider: str, region_name: str):
+        self.cloud_provider = cloud_provider
         self.region_name = region_name
-        self.ec2_client = boto3.client("ec2", region_name=region_name)
-        self.ec2_client_source = boto3.client("ec2", region_name=self.SOURCE_IMAGE_REGION)
-        self.ec2_resource = boto3.resource("ec2", region_name=region_name)
-        self.ec2_resource_source = boto3.resource("ec2", region_name=self.SOURCE_IMAGE_REGION)
         self._ssh_pkey_file = None
 
     @staticmethod
```

Comment:
Let's make it abstract and get rid of raise

```
    def instance_type(test_duration):
        raise NotImplementedError()
```
->
```
@abc.abstractmethod
    def guess_instance_type_from_test_duration(test_duration: int) -> str: ...
```


def instance_type(test_duration: int)